modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -338,6 +338,7 @@ public AbstractTypeScriptClientCodegen() {
         typeMapping.put("Array", "Array");
         typeMapping.put("array", "Array");
         typeMapping.put("boolean", "boolean");
+        typeMapping.put("decimal", "string");
```

Comment:
without this change, https://github.com/OpenAPITools/openapi-generator/blob/6b0e581506549fb0663cd8511be99d5c899bcea8/modules/openapi-generator/src/test/resources/3_0/snakecase-discriminator.yaml#L1556-L1558

for example would exopect a full-blown `Decimal` class, which is never generated. You can see code references to it that are not resolved as the file is not there. For example: https://github.com/OpenAPITools/openapi-generator/blob/ce09134b4877f92fea4e336a2998f69e2cc42ed5/samples/client/petstore/typescript-fetch/builds/snakecase-discriminator/models/FormatTest.ts#L16